Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -122,6 +122,14 @@ public final class SwiftLanguageServer: ToolchainLanguageServer {
   var keys: sourcekitd_keys { return sourcekitd.keys }
   var requests: sourcekitd_requests { return sourcekitd.requests }
   var values: sourcekitd_values { return sourcekitd.values }
+
+  var enablePublishDiagnostics: Bool {
+    // Since LSP 3.17.0, diagnostics can be reported through pull-based requests,
+    // in addition to the existing push-based publish notifications.
+    // If the client supports pull diagnostics, we report the capability
+    // and we should disable the publish notifications to avoid double-reporting.
```

Comment:
Is it documented somewhere that LSP servers shouldn’t publish diagnostics if the client supports `pullDiagnostics`? Or, if not, do you know if other LSP servers are also doing this as well or if VSCode is relying on it?

Reply:
I suggest a solution to this here: https://github.com/apple/sourcekit-lsp/pull/746/files#r1206839125